addons/l10n_ec/models/account_move.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError, ValidationError
+import odoo.addons.decimal_precision as dp
+
+
+class AccountMove(models.Model):
+    _inherit='account.move'
+    
+    def _get_l10n_latam_documents_domain(self):
+        #Filter document types according to ecuadorian move_type
+        domain = super(AccountMove, self)._get_l10n_latam_documents_domain()
+        if self.country_code == 'EC':
+            if self.move_type in ['out_invoice']:
+                domain.extend([('l10n_ec_type', '=', 'out_invoice')])
+            if self.move_type in ['out_refund']:
+                domain.extend([('l10n_ec_type', '=', 'out_refund')])
```

Comment:
@Stentor check if we really need l10n_ec_type or if we can reuse the internal type.

Reply:
We need a mechanism to diferentiate document types according to move type,so the user doesnt use a wrong document type according to the account_move type.

 Exemple:

<record id="ec_04" model="l10n_latam.document.type">
			<field name="code">18</field>
			<field name="name">Factura de Venta</field>
			<field name="report_name">Factura</field>
			<field name="doc_code_prefix">Fact</field>
			<field name="l10n_ec_require_vat">True</field>
			<field name="sequence">15</field>
			<field name="internal_type">invoice</field>
			<field name="l10n_ec_type">out_invoice</field>
			<field name="l10n_ec_authorization">own</field>
			<field name="country_id" ref="base.ec"/>
		</record>

<record id="ec_06" model="l10n_latam.document.type">
			<field name="code">01</field>
			<field name="name">Factura de Compra</field>
			<field name="report_name">Factura</field>
			<field name="doc_code_prefix">Fact</field>
			<field name="l10n_ec_require_vat">True</field>
			<field name="sequence">10</field>
			<field name="internal_type">invoice</field>
			<field name="l10n_ec_type">in_invoice</field>
			<field name="l10n_ec_authorization">third</field>
			<field name="country_id" ref="base.ec"/>
		</record>

The field 10n_ec_require_vat can be removed, though.

Reply:
Check who we did it for Chile, probably we can reuse the same approach.